eth/client.go

```diff
@@ -418,7 +419,186 @@ func (c *client) Bond(amount *big.Int, to ethcommon.Address) (*types.Transaction
+	}
+
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	currentRound, err := c.CurrentRound()
+	if err != nil {
+		return nil, err
+	}
+
+	ep, err := c.GetTranscoderEarningsPoolForRound(sender, currentRound)
+	if err != nil {
+		return nil, err
+	}
+
+	totalBonded := ep.TotalStake
+
+	newStake := totalBonded.Sub(totalBonded, amount)
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	hints := simulateTranscoderPoolUpdate(delegator.DelegateAddress, newStake, transcoders, isFull)
+
+	return c.UnbondWithHint(amount, hints.PosPrev, hints.PosNext)
+}
+
+func (c *client) RebondFromUnbonded(to ethcommon.Address, unbondingLockID *big.Int) (*types.Transaction, error) {
+	sender := c.Account().Address
+
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	currentRound, err := c.CurrentRound()
+	if err != nil {
+		return nil, err
+	}
+
+	ep, err := c.GetTranscoderEarningsPoolForRound(to, currentRound)
+	if err != nil {
+		return nil, err
+	}
+
+	totalBonded := ep.TotalStake
+
+	lock, err := c.GetDelegatorUnbondingLock(sender, unbondingLockID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get unbonding lock")
+	}
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	newStake := totalBonded.Add(totalBonded, lock.Amount)
+
+	hints := simulateTranscoderPoolUpdate(to, newStake, transcoders, isFull)
+
+	return c.RebondFromUnbondedWithHint(to, unbondingLockID, hints.PosPrev, hints.PosNext)
+}
+
+func (c *client) Rebond(unbondingLockID *big.Int) (*types.Transaction, error) {
+	sender := c.Account().Address
+
+	// Get current delegate
+	delegator, err := c.GetDelegator(sender)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get delegator")
+	}
+
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	currentRound, err := c.CurrentRound()
+	if err != nil {
+		return nil, err
+	}
+
+	ep, err := c.GetTranscoderEarningsPoolForRound(sender, currentRound)
+	if err != nil {
+		return nil, err
+	}
+
+	totalBonded := ep.TotalStake
+
+	lock, err := c.GetDelegatorUnbondingLock(sender, unbondingLockID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get unbonding lock")
```

Comment:
Unable -> unable for consistency in error messages

Reply:
fixed in 6c5b9d4d